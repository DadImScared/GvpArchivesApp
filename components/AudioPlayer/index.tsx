import * as React from "react";
import { StyleSheet, View } from "react-native";

import { withAudioPlayer } from "../withAudioPlayer";
import ButtonGroup from "./ButtonGroup";
import DurationSeeker from "./DurationSeeker";
import { IAudioPlayerState } from "./Provider";
import Title from "./Title";

interface IProps {
  audio: IAudioPlayerState;
}

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: "white",
    bottom: 0,
    padding: 8,
    position: "absolute",
    width: "100%",
  }
});

export class AudioPlayer extends React.Component<IProps> {
  render() {
    const {
      audio: {
        isSeeking,
        isPlaying,
        onPlayPausePressed,
        getSongName,
        getSeekSliderPosition,
        getTimeStamp,
        onSeekSliderChange,
        onSeekSliderComplete

      }
    } = this.props;
    return (
      <View style={styles.wrapper}>
        <Title getSongName={getSongName} />
        <DurationSeeker
          isSeeking={isSeeking}
          getSeekSliderPosition={getSeekSliderPosition}
          getTimeStamp={getTimeStamp}
          onSeekSliderChange={onSeekSliderChange}
          onSeekSliderComplete={onSeekSliderComplete}
        />
        <ButtonGroup
          onPlayPausePressed={onPlayPausePressed}
          isPlaying={isPlaying}
        />
      </View>
    );
  }
}

export default withAudioPlayer(AudioPlayer) as any; // property error without "any"
