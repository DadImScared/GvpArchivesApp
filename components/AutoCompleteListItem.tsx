import * as React from "react";
import { Text } from "react-native";

import { ListItem } from "native-base";

interface IProps {
  text: string;
}

export class AutoCompleteListItem extends React.Component<IProps> {

  render() {
    return (
      <ListItem button={true} noIndent={true}>
        <Text>{this.props.text}</Text>
      </ListItem>
    );
  }
}

export default AutoCompleteListItem;
