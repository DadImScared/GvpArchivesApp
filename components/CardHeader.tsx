import * as React from "react";

import { Body, CardItem, Text } from "native-base";
import { routes } from "./SideBar";

interface IProps {
  item: any;
  extraSubHeader?: string;
}

export const CardHeader: React.FunctionComponent<IProps> = ({ item, extraSubHeader }) => (
  <CardItem header={true}>
    <Body>
      <Text>{item.title.trim()}</Text>
      <Text note={true}>{routes[item.category]}{extraSubHeader ? `${extraSubHeader}` : null}</Text>
    </Body>
  </CardItem>
);

export default CardHeader;
