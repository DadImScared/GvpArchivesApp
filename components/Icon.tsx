import * as React from "react";

import { Icon as NativeBaseIcon, NativeBase } from "native-base";

export class Icon extends React.Component<NativeBase.Icon> {
  render() {
    return (
      <NativeBaseIcon {...this.props} />
    );
  }
}

export default Icon;
