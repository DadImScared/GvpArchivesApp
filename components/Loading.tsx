import * as React from "react";

import { Spinner } from "native-base";
import { View } from "react-native";

export const Loading: React.FunctionComponent<{ isLoading: boolean }> = ({ isLoading }) => isLoading ? (
  <View>
    <Spinner />
  </View>
) : null;

export default Loading;
