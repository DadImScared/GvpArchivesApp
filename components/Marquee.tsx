import * as React from "react";

import Ticker, { ITextMarqueeProps } from "react-native-text-ticker";

const ForwardedMarquee = React.forwardRef<Ticker, ITextMarqueeProps>(({ children, ...other }, ref) => (
  <Ticker {...other} ref={ref}>
    {children}
  </Ticker>
));

export default ForwardedMarquee;
