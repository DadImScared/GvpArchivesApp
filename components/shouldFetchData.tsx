import * as React from "react";

import { compose, hoistStatics, lifecycle, withHandlers } from "recompose";
import { Omit } from "utility-types";

interface IProps {
  fetchData: () => void;
  fetchNextPage: () => void;
  isLoading: boolean;
  timeToLive: number;
  nextPage: boolean | string;
  results: any[];
}

interface INextPageHandlerProps extends Omit<IProps, "fetchData"> {
  fetchData: (refresh: boolean) => void;
}

const isValidValidTimeToLive = (time: number) => time ? time >= +new Date() : false;

export const fetchDataHandlers = withHandlers<IProps, { fetchData: () => void }>(() => ({
  fetchData: ({ fetchData, results, isLoading, timeToLive }) => (refresh = false) => {
    if (refresh && !isLoading) {
      fetchData();
    }
    else if (!isLoading && !results.length && !isValidValidTimeToLive(timeToLive)) {
      fetchData();
    }
  }
}));

export const fetchNextPageHandlers = withHandlers<INextPageHandlerProps, { fetchNextPage: () => void }>({
  fetchNextPage: ({ fetchData, fetchNextPage, nextPage, timeToLive, isLoading }) => () => {
    if (isLoading) return;
    if (isValidValidTimeToLive(timeToLive) && typeof nextPage === "string") {
      fetchNextPage();
    }
    else if (!isValidValidTimeToLive(timeToLive)) {
      fetchData(true);
    }
  }
});

const fetchDataLifecycle = lifecycle<IProps, any>({
  componentDidMount() {
    this.props.fetchData();
  }
});

export const shouldFetchData = hoistStatics<IProps>(
  compose(
    fetchDataHandlers,
    fetchNextPageHandlers,
    fetchDataLifecycle
  )
);
