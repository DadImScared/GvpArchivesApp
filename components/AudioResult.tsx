import * as React from "react";

import { Button, Card, CardItem, Left } from "native-base";
import { StyleSheet, Text } from "react-native";

import { Item } from "../reducers/itemsById";
import CardHeader from "./CardHeader";

export interface IAudioResultProps {
  item: Item;
  [key: string]: any;
}

const styles: any = StyleSheet.create({
  button: {
    alignItems: "center",
    justifyContent: "center",
    margin: 4,
    width: "30%"
  }
});

export const AudioResult: React.FC<IAudioResultProps> = ({ item }) => (
  <Card>
    <CardHeader item={item} />
    <CardItem footer={true}>
      <Left>
        <Button primary={true} style={styles.button}><Text style={{ color: "white" }}>Play</Text></Button>
        <Button info={true} style={styles.button}><Text>Playlist</Text></Button>
      </Left>
    </CardItem>
  </Card>
);

export default AudioResult;
