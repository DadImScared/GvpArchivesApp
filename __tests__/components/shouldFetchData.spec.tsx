import * as React from "react";
import * as renderer from "react-test-renderer";

import { createSink } from "recompose";

import { fetchDataHandlers, shouldFetchData } from "../../components/shouldFetchData";

describe("shouldFetchData", () => {
  let props: any;

  beforeEach(() => {
    props = {
      fetchData: jest.fn(),
      fetchNextPage: jest.fn(),
      isLoading: false,
      nextPage: false,
      results: [],
      timeToLive: 0
    };
  });

  it("should fetchData from initial state", () => {
    const sink = createSink((sinkProps: any) => {
      expect(sinkProps).toMatchSnapshot();
    });

    const EnhancedSink = shouldFetchData(sink);
    renderer.create(<EnhancedSink {...props} />);
    expect(props.fetchData).toHaveBeenCalledTimes(1);
  });

  describe("fetchDataHandlers", () => {
    it("should fetchData with refresh passed in as a parameter", () => {
      const sink = createSink((sinkProps: any) => {
        sinkProps.fetchData(true);
        expect(props.fetchData).toHaveBeenCalledTimes(1);
      });

      const EnhancedSink = fetchDataHandlers(sink);
      renderer.create(<EnhancedSink {...props} />);
    });
  });
});
