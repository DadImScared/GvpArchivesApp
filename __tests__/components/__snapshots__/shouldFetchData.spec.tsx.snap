// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shouldFetchData should fetchData from initial state 1`] = `
Object {
  "fetchData": [Function],
  "fetchNextPage": [Function],
  "isLoading": false,
  "nextPage": false,
  "results": Array [],
  "timeToLive": 0,
}
`;
